test/integration/consul-container/libs/consul-cluster/consul-cluster.go

```diff
@@ -0,0 +1,105 @@
+package consul_cluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	consulnode "github.com/hashicorp/consul/integration/consul-container/libs/consul-node"
+)
+
+// Cluster abstract a Consul Cluster by providing
+// a way to create and join a Consul Cluster
+// a way to add nodes to a cluster
+// a way to fetch the cluster leader...
+type Cluster struct {
+	Nodes []consulnode.ConsulNode
+}
+
+// New Create a new cluster based on the provided configuration
+func New(configs []consulnode.Config) (*Cluster, error) {
+	cluster := Cluster{}
+
+	for _, c := range configs {
+		n, err := consulnode.NewConsulContainer(context.Background(), c)
+		if err != nil {
+			return nil, err
+		}
+		cluster.Nodes = append(cluster.Nodes, n)
+	}
+	err := cluster.join()
```

Comment:
It looks like the logic in `join` already exists in`AddNode`. Could we just call that method from here instead?

```suggestion
	nodes := make([]consulnode.ConsulNode, len(configs))
	for idx, c := range configs {
		n, err := consulnode.NewConsulContainer(context.Background(), c)
		if err != nil {
			return nil, err
		}
		nodes[idx] = n
	}
	if err := cluster.AddNodes(nodes); err != nil {
		return nil, err
	}
```